File: solution/array/055-jump-game/solution.java
class Solution {
    public boolean canJump(int[] nums) {
        int farthestReach = 0;
        int lastIndex = nums.length - 1;

        for (int index = 0; index < nums.length; index++) {
            if (index > farthestReach) {
                return false;
            }

            farthestReach = Math.max(
                farthestReach,
                index + nums[index]
            );

            if (farthestReach >= lastIndex) {
                return true;
            }
        }

        return true;
    }
}

File: solution/array/055-jump-game/solution.go
package main

func canJump(nums []int) bool {
	farthestReach := 0
	lastIndex := len(nums) - 1

	for index, jumpLength := range nums {
		if index > farthestReach {
			return false
		}

		if index+jumpLength > farthestReach {
			farthestReach = index + jumpLength
		}

		if farthestReach >= lastIndex {
			return true
		}
	}

	return true
}